extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/WrongAnnotationsProcessor.java

```diff
@@ -0,0 +1,88 @@
+package io.quarkus.arc.deployment;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+
+import io.quarkus.arc.deployment.ValidationPhaseBuildItem.ValidationErrorBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationIndexBuildItem;
+
+public class WrongAnnotationsProcessor {
+
+    @BuildStep
+    void detect(ArcConfig config, ApplicationIndexBuildItem applicationIndex,
+            BuildProducer<ValidationErrorBuildItem> validationError) {
+
+        if (!config.detectWrongAnnotations) {
+            return;
+        }
+
+        IndexView index = applicationIndex.getIndex();
+        List<WrongAnnotation> wrongAnnotations = new ArrayList<>();
+        Function<AnnotationInstance, String> singletonFun = new Function<AnnotationInstance, String>() {
+
+            @Override
+            public String apply(AnnotationInstance annotationInstance) {
+                return String.format("%s declared on %s, use @javax.inject.Singleton instead",
+                        annotationInstance.toString(false), getTargetInfo(annotationInstance));
+            }
+        };
+        wrongAnnotations.add(new WrongAnnotation(DotName.createSimple("com.google.inject.Singleton"), singletonFun));
+        wrongAnnotations.add(new WrongAnnotation(DotName.createSimple("javax.ejb.Singleton"), singletonFun));
+        wrongAnnotations.add(new WrongAnnotation(DotName.createSimple("groovy.lang.Singleton"), singletonFun));
+        wrongAnnotations.add(new WrongAnnotation(DotName.createSimple("jakarta.ejb.Singleton"), singletonFun));
```

Comment:
I'm completely fine with this, but maybe it makes sense to just catch anything other than our CDI Singleton?

Reply:
I don't understand. What do you mean with "to just catch anything"?

Reply:
I mean, if the simple class name is `Singleton` but the full class name is not `javax.inject.Singleton`, throw the exception

Reply:
I think that would be too dangerous and I would prefer we keep a list of the usual suspects.

Reply:
> I mean, if the simple class name is `Singleton` but the full class name is not `javax.inject.Singleton`, throw the exception

We would have to (1) iterate over all annotations in all classes, (2) identify all annotations with simple name == `Singleton` and then find all occurences of those... That might be quite expensive. 

> I think that would be too dangerous and I would prefer we keep a list of the usual suspects.

I agree. In theory, there can be a completely legit `@*.Singleton` used in the app.

Reply:
I don't see how it would be dangerous, but I am perfectly fine with keeping it simple